tools/prune-old-builds.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/python3
+
+# Copyright (C) 2020  Matthew "strager" Glazar
+# See end of file for extended copyright information.
+
+import pipes
+import re
+import os
+import sys
+import shutil
+import argparse
+import subprocess
+from pathlib import Path
+from datetime import datetime
+
+
+FOURTEEN_DAYS = 14
+
+
+def how_old(date: datetime) -> int:
+    """Return a integer that represents how many days has passed given a date"""
+    return (datetime.now() - date).days
+
+
+def error_print(*args, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+
+def get_commits(repo_name: str, repository_url: str) -> list:
+    try:
+        return subprocess.check_output(
+            f"cd {repo_name} \
+            && git fetch --prune {repository_url} '+refs/pull/*/head:refs/remotes/prs/*' '+refs/heads/*:refs/remotes/origin/*' \
+            && git rev-list --all --remotes", shell=True
+        ).decode('utf-8').split('\n')
+    except subprocess.CalledProcessError as err:
+        error_print(err.output.decode())
+        exit(err.returncode)
+
+
+def has_valid_name(folder_name: str) -> bool:
+    return not re.match(r'^[a-f0-9]{40}$', folder_name) == None
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser("Prune old builds")
+    parser.add_argument("repo_name", help="Repository name.", type=str)
+    parser.add_argument(
+        "builds_path",
+        help="The path where the builds are stored.",
+        type=str)
+    parser.add_argument(
+        "repository_url",
+        help="The url to the repository.",
+        type=str)
+    parser.add_argument("-f", "--delete", action="store_true",
+                        help="Delete folders.")
+    args, _ = parser.parse_known_args()
+
+    builds_path = Path(args.builds_path)
+    repository_url = pipes.quote(args.repository_url)
+    repo_name = pipes.quote(args.repo_name)
+
+    if not os.path.exists(builds_path):
+        error_print(
+            f"error: The directory \033[1m{builds_path}\033[0m does not exists.")
+        exit(1)
+
+    if not os.path.exists(repo_name):
+        error_print("Cloning repository")
+        subprocess.check_call(
+            ['git', 'clone', '-q', '--bare', repository_url, repo_name])
```

Comment:
Must fix: `repository_url` and `repo_name` are shell-escaped thus shouldn't be given to `subprocess.check_call` in a list. Use the unescaped string here.